base_name_search_improved/models/ir_model.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# © 2016 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp import SUPERUSER_ID
+
+
+class ModelExtended(models.Model):
+    _inherit = 'ir.model'
+
+    name_search_ids = fields.Many2many(
+        'ir.model.fields',
+        string='Name Search Fields')
+
+    def _register_hook(self, cr, ids=None):
+
+        def make_name_search():
+
+            @api.model
+            def name_search(self, name='', args=None,
+                            operator='ilike', limit=100):
+                # Regular name search
+                res = name_search.origin(
+                    self, name=name, args=args, operator=operator, limit=limit)
+
+                if not res and self._rec_name:
```

Comment:
That should be a rare even, since name_search is seldom called directly with a specific operator.
But it is wise to add that. I can restrict only to `like`, `ilike` and `=`.